// vuestic-ui/dist/es/src/components/va-textarea/VaTextarea.vue_vue_type_script_lang-d8042896.mjs
import { defineComponent, shallowRef, computed } from "vue";
import { p as pick } from "../../../vendor-2b2dbafd.mjs";
import { V as VaInputWrapper } from "../va-input-wrapper/index-66a85aff.mjs";
import { f as focusElement, b as blurElement } from "../../utils/focus-685f6da0.mjs";
import { e as extractComponentProps } from "../../utils/component-options/extract-component-options-6b96d215.mjs";
import { u as useFormFieldProps } from "../../composables/useFormField-3d39d61a.mjs";
import { a as useStatefulProps, b as useStateful } from "../../composables/useStateful-78f75564.mjs";
import { u as useValidationProps, a as useValidationEmits, b as useValidation } from "../../composables/useValidation-d5e122b8.mjs";
import { f as filterComponentProps } from "../../utils/component-options/filter-props-9727421a.mjs";
import { u as useEmitProxy } from "../../composables/useEmitProxy-a435894a.mjs";
const positiveNumberValidator = (val) => {
  if (val > 0 && (val | 0) === val) {
    return true;
  }
  throw new Error(`\`minRows|maxRows\` must be a positive integer greater than 0, but ${val} is provided`);
};
const { createEmits, createListeners } = useEmitProxy([
  "input",
  "change",
  "click",
  "update:modelValue"
]);
const VaInputWrapperProps = extractComponentProps(VaInputWrapper);
const _sfc_main = defineComponent({
  name: "VaTextarea",
  components: { VaInputWrapper },
  props: {
    ...useFormFieldProps,
    ...VaInputWrapperProps,
    ...useStatefulProps,
    ...useValidationProps,
    modelValue: { type: [String, Number], default: "" },
    placeholder: { type: String },
    autosize: { type: Boolean, default: false },
    minRows: {
      type: Number,
      default: 1,
      validator: positiveNumberValidator
    },
    maxRows: {
      type: Number,
      validator: positiveNumberValidator
    },
    resize: {
      type: Boolean,
      default: true
    },
    clearValue: {
      type: [String],
      default: ""
    }
  },
  emits: [...createEmits(), ...useValidationEmits],
  setup(props, { emit }) {
    const textarea = shallowRef();
    const { valueComputed } = useStateful(props, emit, "modelValue", {
      defaultValue: ""
    });
    const focus = () => {
      focusElement(textarea.value);
    };
    const blur = () => {
      blurElement(textarea.value);
    };
    const reset = () => withoutValidation(() => {
      emit("update:modelValue", props.clearValue);
      emit("clear");
      resetValidation();
    });
    const {
      computedError,
      computedErrorMessages,
      listeners: validationListeners,
      validationAriaAttributes,
      isLoading,
      resetValidation,
      withoutValidation
    } = useValidation(props, emit, {
      value: valueComputed,
      focus,
      reset
    });
    const isResizable = computed(() => {
      return props.resize && !props.autosize;
    });
    const computedRowsCount = computed(() => {
      if (!props.autosize) {
        return void 0;
      }
      const rows = valueComputed.value.toString().split("\n").length;
      if (!props.maxRows) {
        return rows;
      }
      return Math.max(props.minRows, Math.min(rows, props.maxRows));
    });
    const computedStyle = computed(() => ({
      resize: isResizable.value ? void 0 : "none"
    }));
    const computedProps = computed(() => ({
      ...pick(props, ["disabled", "readonly", "placeholder", "ariaLabel"])
    }));
    return {
      isResizable,
      validationListeners,
      validationAriaAttributes,
      computedError,
      computedErrorMessages,
      isLoading,
      computedRowsCount,
      valueComputed,
      vaInputWrapperProps: filterComponentProps(VaInputWrapperProps),
      textarea,
      computedStyle,
      listeners: createListeners(emit),
      computedProps,
      focus,
      blur
    };
  }
});
export {
  _sfc_main as _
};
