// vuestic-ui/dist/es/src/components/va-color-input/VaColorInput.vue_vue_type_script_lang-ef5ad674.mjs
import { defineComponent, shallowRef, computed } from "vue";
import { V as VaColorIndicator } from "../va-color-indicator/index-84a92835.mjs";
import { V as VaInput } from "../va-input/index-14c4b30e.mjs";
import { t as throttle } from "../../../vendor-2b2dbafd.mjs";
import { e as extractComponentProps } from "../../utils/component-options/extract-component-options-6b96d215.mjs";
import { u as useStatefulEmits, a as useStatefulProps, b as useStateful } from "../../composables/useStateful-78f75564.mjs";
import { u as useComponentPresetProp } from "../../composables/useComponentPreset-0dfbd395.mjs";
import { u as useTranslation } from "../../composables/useTranslation-0934900c.mjs";
import { f as filterComponentProps } from "../../utils/component-options/filter-props-9727421a.mjs";
const VaInputProps = extractComponentProps(VaInput);
const _sfc_main = defineComponent({
  name: "VaColorInput",
  components: {
    VaInput,
    VaColorIndicator
  },
  emits: [...useStatefulEmits],
  props: {
    ...VaInputProps,
    ...useStatefulProps,
    ...useComponentPresetProp,
    modelValue: { type: String, default: null },
    disabled: { type: Boolean, default: false },
    indicator: {
      type: String,
      default: "dot",
      validator: (value) => ["dot", "square"].includes(value)
    },
    ariaOpenColorPickerLabel: { type: String, default: "$t:openColorPicker" }
  },
  setup: (props, { emit }) => {
    const colorPicker = shallowRef();
    const { valueComputed } = useStateful(props, emit);
    const callPickerDialog = () => {
      var _a;
      return !props.disabled && ((_a = colorPicker.value) == null ? void 0 : _a.click());
    };
    const tabIndexComputed = computed(() => props.disabled ? -1 : 0);
    const inputValue = computed({
      get: () => props.modelValue,
      set: throttle((value) => emit("update:modelValue", value), 500)
    });
    return {
      ...useTranslation(),
      valueComputed,
      inputValue,
      callPickerDialog,
      colorPicker,
      tabIndexComputed,
      vaInputProps: filterComponentProps(VaInputProps)
    };
  }
});
export {
  _sfc_main as _
};
